api/backup.go

```
@@ -54,6 +112,63 @@ func (s *Server) BackupVolumeGet(w http.ResponseWriter, req *http.Request) error
 	return nil
 }
 
+func (s *Server) BackupVolumeUpdateAll(w http.ResponseWriter, req *http.Request) error {
+	var input BackupVolume
+
+	apiContext := api.GetApiContext(req)
+	if err := apiContext.Read(&input); err != nil {
+		return err
+	}
+
+	bvs, err := s.m.ListBackupVolumesSorted()
+	if err != nil {
+		return errors.Wrap(err, "failed to list backup volumes")
+	}
+
+	if input.SyncBackupVolume {
+		for _, bv := range bvs {
+			if _, err := s.m.SyncBackupVolume(bv); err != nil {
+				logrus.Warnf("failed to synchronize backup volume %v", bv.Name)
+			}
+		}
+	}
+
+	apiContext.Write(toBackupVolumeCollection(bvs, apiContext))
+	return nil
+}
+
+func (s *Server) SyncBackupVolume(w http.ResponseWriter, req *http.Request) error {
+	var input SyncBackupVolume
+
+	apiContext := api.GetApiContext(req)
+	if err := apiContext.Read(&input); err != nil {
+		return err
+	}
+
+	volName := mux.Vars(req)["volName"]
+	if volName == "" {
+		return fmt.Errorf("backup volume name is required")
+	}
+
+	bv, err := s.m.GetBackupVolume(volName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get backup volume '%s'", volName)
+	}
+
+	if input.SyncBackupVolume {
+		updatedBackupVolume, err := s.m.SyncBackupVolume(bv)
+		if err != nil {
+			logrus.Warnf("failed to synchronize backup volume %v", volName)
```

Same question as above

---

Yes, let's return the error.